prow/cmd/deck/rerun_test.go

```diff
@@ -232,6 +235,512 @@ func TestRerun(t *testing.T) {
+			if err != nil {
+				t.Fatalf("Error making request: %v", err)
+			}
+			req.AddCookie(&http.Cookie{
+				Name:    "github_login",
+				Value:   tc.login,
+				Path:    "/",
+				Expires: time.Now().Add(time.Hour * 24 * 30),
+				Secure:  true,
+			})
+			mockCookieStore := sessions.NewCookieStore([]byte("secret-key"))
+			session, err := sessions.GetRegistry(req).Get(mockCookieStore, "access-token-session")
+			if err != nil {
+				t.Fatalf("Error making access token session: %v", err)
+			}
+			session.Values["access-token"] = &oauth2.Token{AccessToken: "validtoken"}
+
+			rr := httptest.NewRecorder()
+			mockConfig := &githuboauth.Config{
+				CookieStore: mockCookieStore,
+			}
+			goa := githuboauth.NewAgent(mockConfig, &logrus.Entry{})
+			ghc := &fakeAuthenticatedUserIdentifier{login: tc.login}
+			rc := fakegithub.NewFakeClient()
+			rc.OrgMembers = map[string][]string{"org": {"org-member"}}
+			pca := plugins.NewFakeConfigAgent()
+			jobs := []config.Presubmit{
+				{
+					JobBase: config.JobBase{
+						Name: "whoa",
+						Labels: map[string]string{
+							"foo": "foo",
+						},
+						Annotations: map[string]string{
+							"foo": "foo",
+						},
+						RerunAuthConfig: &prowapi.RerunAuthConfig{
+							AllowAnyone:   false,
+							GitHubUsers:   []string{"authorized", "alsoauthorized"},
+							GitHubTeamIDs: []int{42},
+						},
+					},
+					Brancher: config.Brancher{
+						Branches: []string{"master"},
+					},
+				}}
+			cfg := func() *config.Config {
+				return &config.Config{
+					JobConfig: config.JobConfig{
+						PresubmitsStatic: map[string][]config.Presubmit{"org/repo": jobs},
+					},
+				}
+			}
+			if err := config.SetPresubmitRegexes(jobs); err != nil {
+				t.Fatal(err)
+			}
+			handler := handleRerun(cfg, fakeProwJobClient.ProwV1().ProwJobs("prowjobs"), tc.rerunCreatesJob, authCfgGetter, goa, ghc, rc, &pca, logrus.WithField("handler", "/rerun"))
+			handler.ServeHTTP(rr, req)
+			if rr.Code != tc.httpCode {
+				t.Fatalf("Bad error code: %d", rr.Code)
+			}
+
+			if tc.shouldCreateProwJob {
+				pjs, err := fakeProwJobClient.ProwV1().ProwJobs("prowjobs").List(context.Background(), metav1.ListOptions{})
+				if err != nil {
+					t.Fatalf("failed to list prowjobs: %v", err)
+				}
+				if numPJs := len(pjs.Items); numPJs != 2 {
+					t.Errorf("expected to get two prowjobs, got %d", numPJs)
+				}
+
+			} else if !tc.rerunCreatesJob && tc.httpCode == http.StatusOK {
+				resp := rr.Result()
+				defer resp.Body.Close()
+				body, err := ioutil.ReadAll(resp.Body)
+				if err != nil {
+					t.Fatalf("Error reading response body: %v", err)
+				}
+				var res prowapi.ProwJob
+				if err := yaml.Unmarshal(body, &res); err != nil {
+					t.Fatalf("Error unmarshaling: %v", err)
+				}
+				if res.Spec.Job != "whoa" {
+					t.Errorf("Wrong job, expected \"whoa\", got \"%s\"", res.Spec.Job)
+				}
+				// Label/annotation from new configuration
+				if val, ok := res.Labels["foo"]; !ok {
```

Comment:
Rather than testing individual properties and only doing it in the non-create path - Would it make sense to keep a reference prowjob and compare against that using `cmp.Diff` or some such in both paths? FIelds that are not deterministic (name IIRC?, possibly also creationTimestamp) could be cleared prior to comparing

Reply:
Updated to use this pattern!